sat/cli/bootsys/bos.py

```diff
@@ -277,7 +277,7 @@ def has_completed(self):
             return False
 
         except APIError as err:
```

Comment:
We should look for specifically HTTP 404 responses which indicate that the session has been deleted. Otherwise we are less resilient to other BOS health issues (e.g. if we get an intermittent 503 service unavailable error).